### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by winfred-agbotey**

### CODE REVIEW

File: src/main/java/com/mentorshiptracker/services/privileges/PrivilegeServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private final PermissionRepository permissionRepository;
+    private final RoleRepository roleRepository;
+
+    @Override
+    public Permission createPermission(PermissionDTO permissionDTO) {
+        Permission permission = objectMapper.convertValue(permissionDTO, Permission.class);
+        boolean permissionExists = permissionRepository.existsByNameIgnoreCase(permission.getName());
+        if (!permissionExists) {
+            return permissionRepository.save(permission);
+        } else {
```
Comment (daniel-bontii) [inline src/main/java/com/mentorshiptracker/services/privileges/PrivilegeServiceImpl.java:30]: Remove the else

Comment (winfred-agbotey) [reply src/main/java/com/mentorshiptracker/services/privileges/PrivilegeServiceImpl.java:30]: done

Comment (winfred-agbotey) [reply src/main/java/com/mentorshiptracker/services/privileges/PrivilegeServiceImpl.java:30]: will add it to the next PR

File: src/main/java/com/mentorshiptracker/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfiguration {
```
Comment (daniel-bontii) [inline src/main/java/com/mentorshiptracker/config/SecurityConfiguration.java:20]: "api/admin/**" and "Administrator" are used more than once

Comment (winfred-agbotey) [reply src/main/java/com/mentorshiptracker/config/SecurityConfiguration.java:20]: alright will correct that

File: src/main/java/com/mentorshiptracker/exceptions/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -26,16 +26,18 @@ public final ResponseEntity<ErrorResponse> handlerAllExceptions(Exception ex, We
     }
 
     @ExceptionHandler(UserException.class)
-    public final ResponseEntity<ErrorResponse> handlerUserNotFoundExceptions(Exception ex, WebRequest request) {
+    public final ResponseEntity<ErrorResponse> handlerUserExceptions(Exception ex, WebRequest request) {
```
Comment (daniel-bontii) [inline src/main/java/com/mentorshiptracker/exceptions/GlobalExceptionHandler.java:29]: handleUserException and handlePrevilegeException are identical

You can combine them into one handler using

@ExceptionHandler({UserException.class, PrivilegeException.class})
//rest of method here

Comment (winfred-agbotey) [reply src/main/java/com/mentorshiptracker/exceptions/GlobalExceptionHandler.java:29]: alright

**Pull request merged**